agent/config/builder.go

```diff
@@ -1827,6 +1827,7 @@ func (b *builder) uiConfigVal(v RawUIConfig) UIConfig {
 		MetricsProviderOptionsJSON: stringVal(v.MetricsProviderOptionsJSON),
 		MetricsProxy:               b.uiMetricsProxyVal(v.MetricsProxy),
 		DashboardURLTemplates:      v.DashboardURLTemplates,
+		HCPEnabled:                 os.Getenv("CONSUL_HCP_ENABLED") == "true",
```

Comment:
the only nit that I could think of here is that we probably want a simple unit test to prove that the flag is set when the env var is set.